### PR TITLE
Added support for cosmos blockchain

### PR DESCRIPTION
**Pull request by stanly-johnson**
#### Summary :
Added support for `cosmos:cosmoshub-3` chain to the library

#### Dependencies added :
- [Tendermint Signer](https://github.com/tendermint/sig)

### CODE REVIEW

File: src/blockchains/cosmos.ts
Comment (zachferland) [inline src/blockchains/cosmos.ts]: could we hash (sha256 like eth implementation) the signature output here? so that a fixed length hex string is returned

**Comment by zachferland** [comment]
thanks @stanly-johnson! is this blocking any other work for you? may wait a few days until merging, since we are reorganizing this library a bit at the moment (ref https://github.com/ceramicnetwork/js-3id-blockchain-utils/issues/40)

File: src/blockchains/cosmos.ts
Comment (oed) [inline src/blockchains/cosmos.ts]: Please use the `uint8array` package for base64 etc.

File: src/blockchains/cosmos.ts
Comment (oed) [inline src/blockchains/cosmos.ts:46]: Doesn't seem like this would allow you to sign the tx remotely. The `signTx` function seems to require a wallet instance. For example, this would not work though wallet connect I assume?

**Review by oed — CHANGES_REQUESTED** [review]

File: src/blockchains/cosmos.ts
Comment (oed) [reply src/blockchains/cosmos.ts]: Not sure what `toCanonicalJSONBytes` does, but maybe we can replace it as well?

File: src/blockchains/cosmos.ts
Comment (oed) [inline src/blockchains/cosmos.ts]:
```suggestion
const stringEncode = (str: string): string => uint8arrays.toString(uint8arrays.fromString(str), 'base64pad');
```

File: src/blockchains/cosmos.ts
Comment (stanly-johnson) [reply src/blockchains/cosmos.ts]: Used uint8array for string encoding and removed tendermint/belt. CanonicalJson is meant to normalise JSON to prevent minor changes from breaking the signature. Now that I think about it, its redundant since the JSON is formed within the code itself. Replaced with JSON.stringify()

File: src/blockchains/cosmos.ts
Comment (stanly-johnson) [reply src/blockchains/cosmos.ts:46]: Yes that's correct. I'm not sure any if remote signer implementations exist for cosmos.

File: src/blockchains/cosmos.ts
Comment (oed) [reply src/blockchains/cosmos.ts]: Thanks!

File: src/blockchains/cosmos.ts
Comment (oed) [reply src/blockchains/cosmos.ts:46]: Ok, I would prefer if this was refactored such that any signature logic happens outside of the createLink function with a remote call. Once there is a standard for this type of thing within cosmos, we'll have to update it here to conform to that standard. See also this PR: https://github.com/ceramicnetwork/js-3id-blockchain-utils/pull/46 where we separate out the createLink logic from the verification logic (create should be light on dependencies, validate is not).

Happy to merge this PR once this is fixed here 👍

Comment (stanly-johnson) [reply src/blockchains/cosmos.ts:46]: Sure no problem, I'll make the updates.

**Comment by oed** [comment]
Hey @stanly-johnson are you still intending to finish this? 

Just a heads up that this package has been deprecated in favor of `@ceramicnetwork/blockchain-utils-linking` and `@ceramicnetwork/blockchain-utils-validation` ( in the [js-ceramic repo](https://github.com/ceramicnetwork/js-ceramic)). I believe there should be minimal changes needed in order to port your work to over there.

Please let us know!

**Comment by stanly-johnson** [comment]
@oed Sorry I was a little busy. I will do the migration and finish up other changes this weekend. Looking at the new repo structure, I guess I have to break up this file to two parts, one that goes into linking and other to validation?

**Comment by oed** [comment]
Np @stanly-johnson, thanks for the update. You are right about breaking it up into two parts 👍 Looking forward to the PR!

**Pull request closed without merging**
